multitenancy-spring-boot-autoconfigure/src/main/java/org/axonframework/extensions/multitenancy/autoconfig/TenantPersistentStreamMessageSourceFactory.java

```diff
@@ -0,0 +1,47 @@
+package org.axonframework.extensions.multitenancy.autoconfig;
+
+import io.axoniq.axonserver.connector.event.PersistentStreamProperties;
+import org.axonframework.axonserver.connector.AxonServerConfiguration;
+import org.axonframework.axonserver.connector.event.axon.PersistentStreamMessageSource;
+import org.axonframework.config.Configuration;
+import org.axonframework.extensions.multitenancy.components.TenantDescriptor;
+
+import java.util.concurrent.ScheduledExecutorService;
+
+
+ /**
+ * Factory interface for creating a {@link PersistentStreamMessageSource} for a specific tenant.
+ * The created PersistentStreamMessageSource can be used to read a stream of events from an Axon Server for a specific processor and tenant.
+ * The PersistentStreamMessageSource is configured with the provided processor name, settings, tenant descriptor, and Axon configuration.
+ *
+ * This interface is used to create a {@link PersistentStreamMessageSource} for a given tenant,
+ * @author Stefan Dragisic
+ * @since 4.10
+ */
+@FunctionalInterface
+public interface TenantPersistentStreamMessageSourceFactory {
```

Comment:
I think the factory interface also belongs in another folder, just as the `MultiTenantPersistentStreamMessageSource`. I would suggest to group the together under the aforementioned directory containing other event-specific components.